### ISSUE TITLE
Error when connecting to external redis server

### ISSUE THREAD

**Issue by ismaelrumzan** [instructure/canvas-lms#1651]
Hi

My canvas instance works fine with the local redis. However, when I set it to connect to the external redis, I get the following error in the canvas production log. (I've checked that i can connect to the redis server through the redis-cli - I actually tested with a wrong ip and i get the redis connection error in the canvas production log when I do). I've tested with different versions of redis server in the external and I am always getting the same error below. (The **serverip** below is the internal ip of the server shared between servers - I have removed it)
I am also pasting the redis and cache yml files

**redis.yml**
production:
  servers:
    - redis://serverip:6379
  database: 0

**cache_store.yml**
production:
  cache_store: redis_store
  servers:
   - redis://serverip:6379
  database: 0

**Canvas production log**

[- -] Redis (0.625ms) get settings/encryption_key_hash [serverip:6379]
[- -]   ^[[1m^[[35m (0.8ms)^[[0m  ^[[1m^[[34mSELECT * FROM unnest(current_schemas(false))^[[0m
[- -]   ^[[1m^[[35m (0.8ms)^[[0m  ^[[1m^[[34mSELECT * FROM unnest(current_schemas(false))^[[0m
[- -]   ^[[1m^[[36mSwitchman::Shard Load (1.3ms)^[[0m  ^[[1m^[[34mSELECT "switchman_shards".* FROM "switchman_shards" WHERE "switchman_shards"."database_server_id" = 'test' AND "switchman_shards"."name" IS NULL ORDER BY "switchman_shar$
[- -]   ^[[1m^[[36mSwitchman::Shard Load (0.8ms)^[[0m  ^[[1m^[[34mSELECT "switchman_shards".* FROM "switchman_shards" WHERE "switchman_shards"."database_server_id" = 'development' AND "switchman_shards"."name" IS NULL ORDER BY "switchm$
[- -] Redis (0.738ms) get settings/encryption_key_hash [serverip:6379]
[- -]   ^[[1m^[[35m (0.7ms)^[[0m  ^[[1m^[[34mSELECT * FROM unnest(current_schemas(false))^[[0m
[- -]   ^[[1m^[[35m (0.8ms)^[[0m  ^[[1m^[[34mSELECT * FROM unnest(current_schemas(false))^[[0m
[- -]   ^[[1m^[[36mSwitchman::Shard Load (1.3ms)^[[0m  ^[[1m^[[34mSELECT "switchman_shards".* FROM "switchman_shards" WHERE "switchman_shards"."database_server_id" = 'test' AND "switchman_shards"."name" IS NULL ORDER BY "switchman_shar$
[- -]   ^[[1m^[[36mSwitchman::Shard Load (0.8ms)^[[0m  ^[[1m^[[34mSELECT "switchman_shards".* FROM "switchman_shards" WHERE "switchman_shards"."database_server_id" = 'development' AND "switchman_shards"."name" IS NULL ORDER BY "switchm$
[- -] Redis (0.853ms) get settings/encryption_key_hash [serverip:6379]

**Comment by grahamb** [comment]
There are no errors in the canvas production log snippet you pasted.

**Comment by ismaelrumzan** [comment]
The error happens in passenger where it says invalid key. The issue seems to have been because I was pointing a new server to an existing database copy from another instance that I'm migrating. Seems to be from the encryption key in the settings table not matching the one in the new instance. It did not work even when I changed the key in the table, possibly because of the token. I got it work by running the database migration script. The external redis now works
